### PR TITLE
Document variables

### LINKED ISSUE #68: [Feature request] Support mode for env variables.
Nice work! This is the restclient I'm looking for a long time.

Here is a common case when I test restful api in day to day work. 

Define a env named {{base_url}} in both `production` and `local` modes. So I could switch to `local` to test the code in local server, and switch to `prod` after deployment.

The {{base_url}} may looks like this.

In `production` mode, base_url is `http://xyz.abc/api/v1.0/something`, and in `local` mode base_url is `http://localhost:8080`.

If rest.nvim could support this, I could replace my rest client for good.

### PR DESCRIPTION
**Pull request by klen**
Found this plugin and it's awesome thanks! I was missing fast switching of variables in the document, so I made this feature.
I suppose this will fix #68.

Just an example:
```
# Some API

# Production settings
host: "service.com"
token:  "PRODUCTION_TOKEN"
username: "manager"

# Stage settings
# host: "stage.service.com"
# token:  "STAGE_TOKEN"
# username: "admin"

query: 'where={"query":"test"}'

###

# Get users
GET https://{{host}}/api/users?{{query}}
Authorization: Bearer {{token}}

# Create a user
POST https://{{host}}/api/users
Content-Type: application/json
Authorization: Bearer {{token}}

{
    "name": "{{username}}"
}

```
You may fast change/comment/uncomment the variables you need and keep it with your `http` document.

### CODE REVIEW

**Review by NTBBloodbath — APPROVED** [review]

**Comment by NTBBloodbath** [comment]
Hi, looks great to me (and is even more simple that the solution that I thought :laughing:). Thank you so much!

File: lua/rest-nvim/utils/init.lua [hunk 1]
```diff
@@ -90,15 +90,44 @@ M.read_dynamic_variables = function()
...
+  local first_tree = parser:trees()[1]
+  if not first_tree then return variables end
+
+  local root = first_tree:root()
+  if not root then return variables end
+
+  for node in root:iter_children() do
+    local type = node:type()
+    if type == 'header' then
+      local name = node:named_child(0)
```
Comment (dakennguyen) [inline lua/rest-nvim/utils/init.lua:114]: Hi, thank you for the implementation. 
I see that this PR use node type `header` for local variable section, but treesitter `header`'s name have this pattern `[A-Za-z-]+` ([here](https://github.com/NTBBloodbath/tree-sitter-http/blob/main/src/grammar.json#L722-L725)). It means that we can't use underscore for the variable name, and consider snake_case or UPPERCASE naming convention, I think underscore is necessary
image

Would it be possible to add a new treesitter group for local variables, or if we reuse `header` pattern, could we add underscore?

Comment (NTBBloodbath) [reply lua/rest-nvim/utils/init.lua:114]: Hi, can you please create an issue in the tree-sitter parser repo? In that way we could have a better discussion about what to do. The parser still needs a bit more of work to work in a 1:1 way with rest.nvim features so opening issues in the repo would be great IMO.

Also thank you per ponting this out, I completely missed this :)

Comment (klen) [reply lua/rest-nvim/utils/init.lua:114]: There could be some solutions:

1. Right now you can use CamelCase
```http
contributionID: 456

GET https://test.com/?{{contributionID}}
```
2. We can fix `rest-nvim.utils` to accept "dashed" var names, like this (I may provide PR, just a line of code):
```http
contribution-id: 456

GET https://test.com/?{{contribution-id}}
```
3. We can replace "dashed" var names to underscore, like this (also the small change) btw it's not explicit:
```http
contribution-id: 456

GET https://test.com/?{{contribution_id}}
```

**Pull request merged**